docs/proposals/form-metadata.md

```diff
@@ -0,0 +1,128 @@
+
+```json
+[
+  {
+    "name": "hello-world-db-apb-latest",
+    "metadata": {},
+    "plans": [
+      {
+        "id": "default",
+        "name": "default",
+        "metadata": {
+          "schemas": {
+            "service_instance": {
+              "create": {
+                "form_definition": [
+                  "postgresql_database",
+                  {
+                    "type": "section",
+                    "items": [
+                      "postgresql_user",
+                      {
+                        "key": "postgresql_password",
+                        "type": "password"
+                      }
+                    ]
+                  }
+                ]
+              },
+              "update": {}
+            },
+            "service_binding": {
+              "create": "... same as service_instance.create"
+            }
+          }
+        },
+        "schemas": {
+          "service_instance": {
+            "create": {
+              "parameters": {
+                "$schema": "http://json-schema.org/draft-04/schema",
+                "additionalProperties": false,
+                "properties": {
+                  "postgresql_database": {
+                    "default": "admin",
+                    "title": "PostgreSQL Database Name",
+                    "type": "string"
+                  },
+                  "postgresql_password": {
+                    "default": "admin",
+                    "title": "PostgreSQL Password",
+                    "type": "string"
+                  },
+                  "postgresql_user": {
+                    "default": "admin",
+                    "title": "PostgreSQL User",
+                    "type": "string"
+                  }
+                },
+                "required": [
+                  "postgresql_database",
+                  "postgresql_user",
+                  "postgresql_password"
+                ],
+                "type": "object"
+              }
+            },
+            "update": {}
+          },
+          "service_binding": {
+            "create": "... same as service_instance.create"
+          }
+        }
+      }
+    ]
+  }
+]
+
+```
+
+## Notes
+* This will require changes to [origin-web-catalog](https://github.com/openshift/origin-web-catalog) to read, validate, and use the form definition.  Currently we are only planning on supporting only fieldset/sections for grouping, types, and honoring the order from the form definition.
+
```

Comment:
I think that this can also take care of issue #181 which might be a nice additional feature to adding these params.